playwright/_impl/_locator.py

```diff
@@ -173,12 +172,11 @@ async def element_handle(
         timeout: float = None,
     ) -> ElementHandle:
         params = locals_to_params(locals())
-        return cast(
-            ElementHandle,
-            await self._frame.wait_for_selector(
-                self._selector, strict=True, state="attached", **params
-            ),
+        handle = await self._frame.wait_for_selector(
```

Comment:
I would keep the cast here instead of an additional assert. It gets ensured on the server side already.

Reply:
Since it is assured on server, an assert is much better and cheaper instruction as compared to a redundant func call, and also friendly to type checkers

Reply:
You are right.  From a small experiment assert is 4 times faster compared to a cast.